###  **Introduction**

This project shows how to build a simple **binary classification model** using **TensorFlow and Keras**. We use **`make_classification()`** from scikit-learn to create a small dataset with two features and two output classes.

The data is then **standardized** using **`StandardScaler`** to help the model learn better. A simple **neural network** with one layer and a **sigmoid activation** is trained to predict the class of each sample.

This project is an easy example to understand the basic steps of a machine learning workflow — **data creation, preprocessing, model building, training, and evaluation**.


**Import necessary libraries**

In [1]:
import tensorflow as tf
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

**Create synthetic dataset**

We will create a simple 2-feature synthetic binary-classification dataset for our demonstration and then split it into training and testing.

In [2]:
X, y = make_classification(
    n_samples=1000,
    n_features=2,
    n_informative=2,
    n_redundant=0,
    n_repeated=0,
    n_classes=2,
    random_state=42
)



In [3]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

**Standardize the Dataset**

In [4]:
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

**Building a neural network**

In [5]:
model = tf.keras.Sequential([
    tf.keras.layers.Dense(1, activation='sigmoid', input_shape=(2,))
])

/usr/local/lib/python3.12/dist-packages/keras/src/layers/core/dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


**Compile the Model**

In [6]:
model.compile(optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy'])

**Train the Model**

In [7]:
history = model.fit(X_train, y_train,
                    epochs=50,
                    batch_size=16,
                    validation_split=0.1,
                    verbose=0)

**Model Evaluation**

In [8]:
loss, accuracy = model.evaluate(X_test, y_test, verbose=0)

In [9]:
accuracy

0.875

### 🧠 **Conclusion**

In this project, we built a simple neural network using **TensorFlow and Keras** to perform binary classification on a synthetic dataset generated with **`make_classification()`**.

After preprocessing the data with **StandardScaler** and training the model for 50 epochs, the network achieved a **test accuracy of 87.5%**, demonstrating that even a minimal model with just one dense neuron and sigmoid activation can effectively separate two classes when the data is well-structured.

This experiment highlights:

* The importance of **data normalization** for faster convergence.
* The effectiveness of **binary cross-entropy** loss for classification problems.
* How even a **simple neural network** can achieve strong performance on linearly separable data.

Overall, this project provides a clear and practical introduction to binary classification using neural networks in TensorFlow.
